app/panel/components/Header.jsx

```diff
@@ -134,8 +135,13 @@ class Header extends React.Component {
 		);
 	}
 
-	clickLogo = () => {
-		this.props.history.push(this.props.is_expert ? '/detail/blocking' : '/');
+	determineBackPath = () => {
+		const { entries, location } = this.props.history;
+		if (location.pathname === '/stats' && (entries.length > 1 &&
+		entries[entries.length - 2].pathname.slice(0, 13) === '/subscription')) {
```

Comment:
@Eden12345 let's make this a regular expression match to make our intentions a bit more clear.

Reply:
@jsignanini done!